Updates to tabledemo and shared autonomy

iamwolf opened the pull request:
- **Added shortcut F10 to open the task panel**, similar to F8, F9, and Ctrl+B - since we're finding ourselves opening the task panel a lot. Happy to change to another key, but F10 came in handy after F8, F9
- **Methods to jump in the task tree and continue** - in tabledemo used to repeat the spiel until there are no more objects that have been segmented (clusterObjects is empty). If there's a better way than pausing and continuing, please let me know :)
- **Fix for EXOTica planning when checking manip plan info**: EXOTica mirrors the 0 plan info for successful requests and 13 when constraints are not satisfied, but we also have negative return codes (this shouldn't conflict with Matlab planning - does it?)
- Minor changes that are flagged with the fixed base switch

iamwolf commented:
@patmarion @mauricefallon are these changes good to be merged?

patmarion commented on src/python/ddapp/tasks/robottasks.py:
```diff
@@ -178,7 +178,7 @@ def getDefaultProperties(properties):
 
     def run(self):
-        if robotSystem.ikPlanner.lastManipPlan and max(robotSystem.ikPlanner.lastManipPlan.plan_info) <= 10:
+        if robotSystem.ikPlanner.lastManipPlan and max(robotSystem.ikPlanner.lastManipPlan.plan_info) <= 10 and min(robotSystem.ikPlanner.lastManipPlan.plan_info) >= 0:
```
I'm not familiar with this, is this saying that sometimes an infeasible plan_info is < 0?

iamwolf replied:
Yes, exactly -- is this a problem for drake or the current system?

patmarion commented on src/python/ddapp/tabledemo.py:
```diff
@@ -1194,6 +1196,14 @@ def onPropertyChanged(self, propertySet, propertyName):
+    def pickupMoreObjects(self):
+        if len(self.tableDemo.clusterObjects) > 0: # There is still sth on the table, let's do it again!
+            print "There are more objects on the table, going at it again!"
+            self.taskTree.selectTaskByName('reach')
```
I think this is probably working very well for you.  It's also possible to make a task that adds new tasks to the queue.  The result would be that list of tasks grows each loop, instead of returning to the start, which may or may not be desirable.

patmarion replied:
I think your looping strategy is good.  The only issue is that it has to know the name of the starting task, 'reach' in this case, but I think that's fine.

patmarion commented:
+1 looks good.  Only thing I'd be worried about is the new check for negative plan_info.  The problem is that there are a few other places in the code that check plan_info, and have hard coded assumptions about what is/isn't feasible.  For example, in playbackpanel.py the functions
isPlanFeasible() and getPlanInfo().  Maybe we could move the isPlanFeasible() check to a single reusable utility function in a common module.

iamwolf commented:
@patmarion Thanks for the feedback. I've checked and the same assumption is used in playbackpanel as you mentioned as well as in valvedemo. I've changed it in both places as well and added it to this PR. I think moving to a global function in the mid-term is a good idea. (happy to add it to my todo list; we can also write those assumptions down somewhere in a wiki or so in case somebody else wants to implement or expand on external planning)

patmarion commented:
Thanks!

patmarion commented:
oops, looks like a small typo:

  File "/home/pat/source/drc/drc/software/build/lib/python2.7/dist-packages/ddapp/playbackpanel.py", line 261
    return plan is not None and max(plan.plan_info) < 10 and min(plan.plan_info) >= 0:

iamwolf commented:
Maurice just caught that as well and will submit the fix with #78 - apologies for overlooking this
